src/components/ManagingHouseForm.vue

```diff
@@ -0,0 +1,314 @@
+        <InputText
+          id="houseBlock"
+          placeholder="Блок"
+          v-model.trim="houseData.address.houseBlock"
+          :class="{
+            'p-invalid': v$.houseData.address.houseBlock.$error,
+          }"
+          @blur="v$.houseData.address.houseBlock.$touch"
+        />
+        <small v-if="v$.houseData.address.houseBlock.$error" class="p-error" id="houseBlock-help">{{
+          v$.houseData.address.houseBlock.$errors[0].$message
+        }}</small>
+      </p>
+      <p>
+        <label class="dialog-item dialog-item-address" for="houseNumber">Номер будинку: </label>
+        <InputText
+          id="houseNumber"
+          placeholder="Номер"
+          v-model.trim="houseData.address.houseNumber"
+          :class="{
+            'p-invalid': v$.houseData.address.houseNumber.$error,
+          }"
+          @blur="v$.houseData.address.houseNumber.$touch"
+        />
+        <small v-if="v$.houseData.address.houseNumber.$error" class="p-error" id="houseNumber-help">{{
+          v$.houseData.address.houseNumber.$errors[0].$message
+        }}</small>
+      </p>
+      <p>
+        <label class="dialog-item dialog-item-address" for="zipCode">Код : </label>
+        <InputText
+          id="zipCode"
+          placeholder="Код"
+          v-model.trim="houseData.address.zipCode"
+          :class="{
+            'p-invalid': v$.houseData.address.zipCode.$error,
+          }"
+          @blur="v$.houseData.address.zipCode.$touch"
+        />
+        <small v-if="v$.houseData.address.zipCode.$error" class="p-error" id="zipCode-help">{{
+          v$.houseData.address.zipCode.$errors[0].$message
+        }}</small>
+      </p>
+    </div>
+    <div class="buttons-container">
+      <Button
+        label="Зберегти зміни"
+        icon="pi pi-check"
+        autofocus
+        class="p-button-info"
+        type="submit"
+        :disabled="v$.houseData.$invalid"
+        id="add-new-house-btn"
+      />
+      <Button
+        label="Скасувати зміни"
+        icon="pi pi-times"
+        @click="cancelManagingModal"
+        class="p-button-outlined p-button-info"
+        id="cancel-btn"
+      />
+    </div>
+  </form>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import useVuelidate from '@vuelidate/core';
+import Button from 'primevue/button';
+import InputText from 'primevue/inputtext';
+import { houseValidations } from '@/utils/house-validations';
+import { AddressInterface, HousesActionsEnum } from '@/store/houses/types';
+import { StoreModuleEnum } from '@/store/types';
+import { HouseModel } from '@/shared/models/house.model';
+
+export default defineComponent({
+  name: 'ManagingHouseForm',
+  components: {
+    Button,
+    InputText,
+  },
+  props: {
+    cooperationId: {
+      type: Number,
+      required: true,
+    },
+    houseData: {
+      type: HouseModel,
+      required: true,
+    },
+    isEditHouse: {
```

Comment:
You don't need this prop. You can check if house data has an ID, than this is model for editing, if there's no ID - this is new model.

Reply:
All right)